api.js

```diff
@@ -0,0 +1,230 @@
+	this.exceptionCount++;
+	err.type = 'exception';
+	this.emit('error', err);
+	this.errors.push(err);
+};
+
+Api.prototype._handleStats = function (stats) {
+	this.testCount += stats.testCount;
+};
+
+Api.prototype._handleTest = function (test) {
+	test.title = this._prefixTitle(test.file) + test.title;
+
+	var isError = test.error.message;
+
+	if (isError) {
+		this.errors.push(test);
+	} else {
+		test.error = null;
+	}
+
+	this.emit('test', test);
+};
+
+Api.prototype._prefixTitle = function (file) {
+	if (this.fileCount === 1) {
+		return '';
+	}
+
+	var separator = ' ' + chalk.gray.dim(figures.pointerSmall) + ' ';
+
+	var base = path.dirname(this.files[0]);
+
+	if (base === '.') {
+		base = this.files[0] || 'test';
+	}
+
+	base += path.sep;
+
+	var prefix = path.relative('.', file)
+		.replace(base, '')
+		.replace(/\.spec/, '')
+		.replace(/test\-/g, '')
+		.replace(/\.js$/, '')
+		.split(path.sep)
+		.join(separator);
+
+	if (prefix.length > 0) {
+		prefix += separator;
+	}
+
+	return prefix;
+};
+
+Api.prototype.run = function () {
+	var self = this;
+
+	return handlePaths(this.files)
+		.map(function (file) {
+			return path.resolve(file);
+		})
+		.then(function (files) {
+			if (files.length === 0) {
+				return Promise.reject(new Error('Couldn\'t find any files to test'));
+			}
+
+			self.fileCount = files.length;
+
+			var tests = files.map(self._runFile);
+
+			// receive test count from all files and then run the tests
+			var statsCount = 0;
+			var deferred = Promise.pending();
+
+			tests.forEach(function (test) {
+				var counted = false;
+
+				function tryRun() {
+					if (counted) {
+						return;
+					}
+
+					if (++statsCount === self.fileCount) {
+						self.emit('ready');
+
+						var method = self.serial ? 'mapSeries' : 'map';
+
+						deferred.resolve(Promise[method](files, function (file, index) {
+							return tests[index].run();
+						}));
+					}
+				}
+
+				test.on('stats', tryRun);
+				test.catch(tryRun);
+			});
+
+			return deferred.promise;
+		})
+		.then(function (results) {
+			// assemble stats from all tests
+			self.stats = results.map(function (result) {
+				return result.stats;
+			});
+
+			self.tests = results.map(function (result) {
+				return result.tests;
+			});
+
+			self.tests = flatten(self.tests);
+
+			self.passCount = sum(self.stats, 'passCount');
+			self.failCount = sum(self.stats, 'failCount');
+		});
+};
+
+function handlePaths(files) {
+	if (files.length === 0) {
+		files = [
+			'test.js',
+			'test-*.js',
+			'test/*.js'
+		];
+	}
+
+	files.push('!**/node_modules/**');
+
+	// convert pinkie-promise to Bluebird promise
+	files = Promise.resolve(globby(files));
+
+	return files
+		.map(function (file) {
+			if (fs.statSync(file).isDirectory()) {
+				return handlePaths([path.join(file, '*.js')]);
+			}
+
+			return file;
+		})
+		.then(flatten)
+		.filter(function (file) {
+			return path.extname(file) === '.js' && path.basename(file)[0] !== '_';
```

Comment:
How do .jsx test files run?
@vdemedes 

Debugged to find this, but unfortunately/luckily this file has been refactored by @jamestalmage 
Hoping that its fixed in next release or minor release to patch this is released soon.